### PR TITLE
refactor: Removes usage of redux (siding with widely adopted conventions in which implementer manages sidebar show/hide state)

### PR DESCRIPTION
**Pull request by schne324**

### CODE REVIEW

File: src/lib/components/Toast/index.js [hunk 1]
```diff
@@ -62,7 +62,9 @@ export default class Toast extends Component {
     }
   }
 
-  componentWillReceiveProps(nextProps) {
+  // TODO: getting rid of the destroy state all together would simplify this
+  // (if props.show changes => do the show/hide stuff)
+  UNSAFE_componentWillReceiveProps(nextProps) {
```
Comment (schne324) [inline src/lib/components/Toast/index.js]: see #28

**Comment by schne324** [comment]
I apologize for the nasty diffs.  In removing dependencies I managed to update to a newer version of react and, as a result, I removed all but 1 use of UNSAFE lifecycle methods.  Leaving the [toast refactor](#28) for a separate PR

File: demo/index.js [hunk 2]
```diff
@@ -23,91 +23,125 @@ import Checkbox from './patterns/components/Checkbox';
...
+
+    if (show && this.topBarTrigger) {
+      this.topBarTrigger.focus();
+    }
+
+    this.setState({ show: !show });
+  }
+
+  render() {
+    const SideBarLink = ({ to, children }) => (
```
Comment (stephenmathieson) [inline demo/index.js]: Any reason this is being defined within another component's `render` method?

Comment (schne324) [reply demo/index.js]: nope, good catch

**Pull request merged**